File: TuduManayer/java/TuduManager/domain/src/main/java/net/seymourpoler/tudumanager/domain/todo/search/SearchTodoService.java
package net.seymourpoler.tudumanager.domain.todo.search;

import net.seymourpoler.tudumanager.domain.todo.search.models.Todo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class SearchTodoService implements ISearchTodoService {

    private final ISearchTodoRepository repository;

    @Autowired
    public SearchTodoService(ISearchTodoRepository repository) {
        this.repository = repository;
    }

    @Override
    public List<Todo> search(String searchText) {

        if(isNotValid(searchText)){
            return List.of();
        }

        return repository.search(searchText);
    }

    private boolean isNotValid(String searchText) {
        return searchText == null || isWhiteSpace(searchText);
    }

    private boolean isWhiteSpace(String searchText) {
        return searchText.replaceAll("\\s+", "").isEmpty();
    }
}

File: TuduManayer/dotnet/react.mvc-core.postgres.dapper/TuduManager/src/TuduManayer.Domain/Todo/Search/SearchTodoService.cs
using System.Collections.Generic;

namespace TuduManayer.Domain.Todo.Search
{
    public interface ISearchTodoService
    {
        IReadOnlyCollection<Models.Todo> Search(string searchText);
    }

    public class SearchTodoService : ISearchTodoService
    {
        private readonly ISearchTodoRepository repository;

        public SearchTodoService(ISearchTodoRepository repository)
        {
            this.repository = repository;
        }

        public IReadOnlyCollection<Models.Todo> Search(string searchText)
        {
            throw new System.NotImplementedException();
        }
    }
}